From ded3d6640f321766c6db1f168f16e9dbfe2e6abd Mon Sep 17 00:00:00 2001
From: lilijun <lilijun@eswincomputing.com>
Date: Fri, 7 Feb 2025 17:46:07 +0800
Subject: [PATCH 409/551] fix:Dvb and z530 boot failed issue

Changelogs:
Do not include "eswin-win2030-die0-soc.dtsi" twice.

Signed-off-by: lilijun <lilijun@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-dvb-dsi-panel.dtsi  | 5 +++--
 arch/riscv/boot/dts/eswin/eic7700-z530-dsi-panel.dtsi | 4 +++-
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-dvb-dsi-panel.dtsi b/arch/riscv/boot/dts/eswin/eic7700-dvb-dsi-panel.dtsi
index 61514cfcce99..e520a4e89746 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-dvb-dsi-panel.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7700-dvb-dsi-panel.dtsi
@@ -17,8 +17,6 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
-#include "eswin-win2030-die0-soc.dtsi"
-
 &SOC {
 	panel_disp: panel_disp {
 		compatible = "raspberrypi,7inch-dsi";
@@ -46,6 +44,9 @@ dc_out_dsi: endpoint@0 {
 
 &dsi_controller {
 	status = "okay";
+	dependencies = <&reg_display>;
+	dependencies = <&toshiba_bridge>;
+	dependencies = <&panel_disp>;
 	ports {
 		#address-cells = <1>;
 		#size-cells = <0>;
diff --git a/arch/riscv/boot/dts/eswin/eic7700-z530-dsi-panel.dtsi b/arch/riscv/boot/dts/eswin/eic7700-z530-dsi-panel.dtsi
index 2875ce48c1a8..dfa297dfc44f 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-z530-dsi-panel.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7700-z530-dsi-panel.dtsi
@@ -17,7 +17,6 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
-#include "eswin-win2030-die0-soc.dtsi"
 
 &SOC {
 	panel_disp: panel_disp {
@@ -46,6 +45,9 @@ dc_out_dsi: endpoint@2 {
 
 &dsi_controller {
 	status = "okay";
+	dependencies = <&reg_display>;
+	dependencies = <&panel_disp>;
+	dependencies = <&toshiba_bridge>;
 	ports {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
2.50.1

